proto/backend.proto

```diff
@@ -306,3 +306,164 @@ message StreamPacket {
   // JSON-encoded data to publish into a channel.
   bytes data = 1;
 }
+
+//-----------------------------------------------------------------
+// InstanceSettings -- EXPERIMENTAL and is subject to change until 12.??
+//-----------------------------------------------------------------
+
+service InstanceSettings {
+  // This allows verifying the app/datasource settings before saving
+  // This is a specalized form of the validation/mutation hooks that only work for instance settings
+  // The response will be in the preferred storage apiVersion
+  rpc ProcessInstanceSettings(ProcessInstanceSettingsRequest) returns (ProcessInstanceSettingsResponse);
+}
+
+message ProcessInstanceSettingsRequest {
+  // Unlike normal plugin context, this may not be saved and the lastUpdatedMS will be changed
+  // with each request.  
+  PluginContext pluginContext = 1;
+
+  // Operation is the type of resource operation being checked for admission control
+  // https://github.com/kubernetes/kubernetes/blob/v1.30.0/pkg/apis/admission/types.go#L158
+  enum Operation {
+    CREATE = 0;
+    UPDATE = 1;
+    DELETE = 2;
+  }
+  
+  Operation operation = 2;
+  
```

Comment:
nit: field number 3?

Reply:
I don't think so -- Operation is a type definition, not a field, it becomes:
```
ProcessInstanceSettingsRequest_Operation
```

Reply:
Sorry, I meant that field 3 appears unused

Reply:
🤦🏻 😄